e2e/test_operator_ha/operator_ha_test.go

```diff
@@ -211,4 +213,105 @@ var _ = Describe("Operator HA tests", Label("e2e", "pr"), func() {
 			}).WithTimeout(10 * time.Minute).WithPolling(2 * time.Second).Should(BeNumerically(">=", desiredRunningPods))
 		})
 	})
+
+	When("locality based exclusions are enabled", func() {
+		var initialUseLocalitiesForExclusion bool
+
+		BeforeEach(func() {
+			spec := fdbCluster.GetRemote().GetCluster().Spec.DeepCopy()
+			initialUseLocalitiesForExclusion = fdbCluster.GetRemote().GetCluster().UseLocalitiesForExclusion()
+			spec.AutomationOptions.UseLocalitiesForExclusion = pointer.Bool(true)
+			fdbCluster.GetRemote().UpdateClusterSpecWithSpec(spec)
+			Expect(fdbCluster.GetRemote().GetCluster().UseLocalitiesForExclusion()).To(BeTrue())
+		})
+
+		AfterEach(func() {
+			spec := fdbCluster.GetRemote().GetCluster().Spec.DeepCopy()
+			spec.AutomationOptions.UseLocalitiesForExclusion = pointer.Bool(initialUseLocalitiesForExclusion)
+			fdbCluster.GetRemote().UpdateClusterSpecWithSpec(spec)
+		})
+
+		When("when a remote log has network latency issues and gets replaced", func() {
+			var experiment *fixtures.ChaosMeshExperiment
+
+			BeforeEach(func() {
+				dcID := fdbCluster.GetRemote().GetCluster().Spec.DataCenter
+
+				status := fdbCluster.GetPrimary().GetStatus()
+
+				var processGroupID fdbv1beta2.ProcessGroupID
+				for _, process := range status.Cluster.Processes {
+					dc, ok := process.Locality[fdbv1beta2.FDBLocalityDCIDKey]
+					if !ok || dc != dcID {
+						continue
+					}
+
+					var isLog bool
+					for _, role := range process.Roles {
+						if role.Role == "log" {
+							isLog = true
+							break
+						}
+					}
+
+					if !isLog {
+						continue
+					}
+
+					processGroupID = fdbv1beta2.ProcessGroupID(process.Locality[fdbv1beta2.FDBLocalityInstanceIDKey])
+					break
+				}
+
+				log.Println("Will inject chaos into", processGroupID, "and replace it")
+				var replacedPod corev1.Pod
+				for _, pod := range fdbCluster.GetRemote().GetLogPods().Items {
+					if fixtures.GetProcessGroupID(pod) != processGroupID {
+						continue
+					}
+
+					replacedPod = pod
+					break
+				}
+
+				log.Println("Inject latency chaos")
+				experiment = factory.InjectNetworkLatency(
+					fixtures.PodSelector(&replacedPod),
+					chaosmesh.PodSelectorSpec{
+						GenericSelectorSpec: chaosmesh.GenericSelectorSpec{
+							Namespaces:     []string{fdbCluster.GetRemote().Namespace()},
```

Comment:
Discussed offline, we also want to have a test where the primary and remote side have latencies injected.